core-web/libs/sdk/client/src/lib/client/content/shared/types.ts

```diff
@@ -24,49 +26,12 @@ export type SortBy = {
  */
 export type BuildQuery = (qb: QueryBuilder) => Equals;
 
-/**
- * Main fields of a Contentlet (Inherited from the Content Type).
- */
-export interface ContentTypeMainFields {
-    hostName: string;
-    modDate: string;
-    publishDate: string;
-    title: string;
-    baseType: string;
-    inode: string;
-    archived: boolean;
-    ownerName: string;
-    host: string;
-    working: boolean;
-    locked: boolean;
-    stInode: string;
-    contentType: string;
-    live: boolean;
-    owner: string;
-    identifier: string;
-    publishUserName: string;
-    publishUser: string;
-    languageId: number;
-    creationDate: string;
-    url: string;
-    titleImage: string;
-    modUserName: string;
-    hasLiveVersion: boolean;
-    folder: string;
-    hasTitleImage: boolean;
-    sortOrder: number;
-    modUser: string;
-    __icon__: string;
-    contentTypeIcon: string;
-    variant: string;
-}
-
 /**
  * The contentlet has the main fields and the custom fields of the content type.
  *
  * @template T - The custom fields of the content type.
  */
-export type Contentlet<T> = T & ContentTypeMainFields;
+export type Contentlet<T> = T & DotCMSBasicContentlet;
```

Comment:
Do we need this `Contentlet` type?

Reply:
We need it to maintain the generics

Reply:
I see that we have it in the types lib. Removed!